server/controllers/admin/roles.js

```diff
@@ -151,6 +141,71 @@ function listForUser(req, res, next) {
     .done();
 }
 
+
+function rolesAction(req, res, next) {
+
+  const roleUuid = db.bid(req.params.roleUuid);
+  const sql = `
+    SELECT a.id, a.description, IFNULL(s.affected, 0) as affected
+    FROM actions a
+    LEFT JOIN(
+      select  actions_id , 1 as affected 
+      FROM role_actions ra
+      JOIN role ro ON ra.role_uuid = ro.uuid
+      WHERE ro.uuid = ?
+    )s ON s.actions_id = a.id
+  `;
+
+  db.exec(sql, [roleUuid])
+    .then((actions) => {
+      res.json(actions);
+    })
+    .catch(next)
+    .done();
+}
+
+
+// affect roles to a user
+// roles ares permissions
+function assignActionToRole(req, res, next) {
+  const data = req.body;
+
+  const actionIds = [].concat(data.action_ids);
+
+  const roleUuid = db.bid(data.role_uuid);
+
+
+  const deleleUserRoles = `DELETE FROM role_actions WHERE role_uuid=? `;
+  const addAction = `INSERT INTO role_actions SET ?`;
+
+  db.exec(deleleUserRoles, roleUuid)
+    .then(() => {
+      const promisses = actionIds.map(actionId => {
```

Comment:
If you are going to add a ton of promises here, it might be a good idea to wrap this all in a `db.transaction()`.  That way, if connection to the database is lost for some reason we do not end up with half the data in an undefined state.

Reply:
okay, good idea